main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/beamx"
+)
+
+var (
+	input_text = flag.String("input-text", "Default input text", "Input text to print.")
+)
+
+type Result = struct {
+	Pipeline    *beam.Pipeline
+	Scope       beam.Scope
+	PCollection beam.PCollection
+}
+
+func my_pipeline(input_text string) Result {
+	beam.Init()
+
+	pipeline, scope := beam.NewPipelineWithRoot()
+
+	elements := beam.Create(scope, "Hello", "World!", input_text)
+	elements = beam.ParDo(scope, func(elem string, emit func(string)) { fmt.Println(elem); emit(elem) }, elements)
```

Comment:
While an anonymous function *can* work in Beam Go, it will cause more trouble than it's worth. Move it to a standard definition, and use an init block to register the function.

Otherwise, folks moving beyond the direct runner will have a bad time.

```
func printAndEmit(elem string, emit(string)) { 
  fmt.Println(elem)
  emit(elem) 
}

func init() {
  // DoFns should be registered with Beam.
  beam.RegisterFunction(printAndEmit)
}
```

Reply:
That's good to know! Done.